src/hooks/usePokemon/usePokemon.js

```diff
@@ -11,17 +11,74 @@ const usePokemonName = () => {
   }
 }
 
+const usePokemonVariants = () => {
+  const [pokemonVariants, setPokemonVariants] = useGlobalState('pokemon-variants', {})
+  const { pokemonName } = usePokemonName()
+  useEffect(async () => {
+    if (pokemonVariants[pokemonName]) {
+      return null;
+    }
+    setPokemonVariants({...pokemonVariants, [pokemonName]: "LOADING"})
+    const pokemonSpeciesObject = await PokeAPI.getPokemonSpeciesByName(pokemonName.toLowerCase())
+    if (!pokemonSpeciesObject) {
+      return null;
+    }
+    setPokemonVariants({...pokemonVariants, [pokemonName]: pokemonSpeciesObject.varieties})
+  }, [pokemonName])
+  return { pokemonVariants }
+}
+
+const usePokemonVariation = () => {
+  const [pokemonVariation, setPokemonVariation] = useGlobalState('pokemon-variation', {})
+  const { pokemonName } = usePokemonName()
+  const onUpdatePokemonVariation = (index) => (event) => {
+    const varient = event.target.value;
+    setPokemonVariation({
+      ...pokemonVariation, [index]: varient
+    })
+  }
+
+  useEffect(() => {
+    if (pokemonVariation[0]) {
+      setPokemonVariation({
+        ...pokemonVariation, 0: null // instead of 0, reset index of pokemonName when it's indexed
```

Comment:
maybe move this comment up, or better explain what this useEffect is for?

Reply:
It's essentially for clearing your variation choice if you change your pokemonName (effectively you want a different pokemon).

I don't know where the best place to put that info is.... or if I can label useEffects

Reply:
just a comment above the useEffect would be good